pkg/baremetal/v1alpha3/types.go

```diff
@@ -74,9 +79,14 @@ type Addon struct {
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-type BareMetalMachineProviderSpec struct {
-	metav1.TypeMeta `json:",inline"`
+type BareMetalMachine struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
 
+	Spec BareMetalMachineSpec `json:"spec,omitempty"`
```

Comment:
I think we should also include `Status` on our machine as per: https://cluster-api.sigs.k8s.io/developer/architecture/controllers/machine.html